Neos.ContentRepository/Classes/ViewHelpers/Widget/Controller/PaginateController.php

```diff
@@ -121,7 +121,7 @@ public function indexAction($currentPage = 1)
         }
         $offset = ($this->currentPage > 1) ? (integer)($itemsPerPage * ($this->currentPage - 1)) : null;
         if ($this->parentNode === null) {
-            $nodes = array_slice($this->nodes, $offset, $itemsPerPage);
+            $nodes = (array) array_slice($this->nodes, $offset, $itemsPerPage);
```

Comment:
i would much rather change it here in: initializeAction()

`$this->nodes = $this->widgetConfiguration['nodes'];`

something like `if $this->widgetConfiguration['nodes'] instaceof TraversableNodes` ...

Reply:
also by reading your provided excepfion - your fix wont work i assume. youd need to "transform" the first param to an array but via getIterator()->getArrayCopy() or toArray()